src/panel/service_panel.js

```diff
@@ -1,47 +1,81 @@
-import ServicePanelView from '../views/service_panel.dot';
-import Panel from '../libs/panel';
-import nconf from '@qwant/nconf-getter';
+/* globals _ */
+import React from 'react';
+import ReactDOM from 'react-dom';
 import CategoryService from '../adapters/category_service';
 
-export default class ServicePanel {
-  constructor() {
-    this.panel = new Panel(this, ServicePanelView);
-    this.categories = CategoryService.getCategories();
-    this.mustDeployCategories = this.categories.length > 8;
-    this.isDeployed = false;
-    this.isDirectionActive = nconf.get().direction.enabled;
-    this.active = true;
-  }
+// Returns a function, which when called will navigate to the url
+const navTo = url => () => {
+  window.app.navigateTo(url);
+};
 
-  toggleCategories() {
-    this.isDeployed = !this.isDeployed;
-    this.panel.update();
-  }
+const CategoryButton = ({ category }) => {
```

Comment:
There are two ways of defining React components (see [React doc](https://reactjs.org/docs/components-and-props.html)): 
- the simpler way is like here, as a function which takes properties (called *props* in React) as fields in its first argument, and returns HTML.
- another way is below, as an ES6 class. It must have at least a `render` function which returns HTML. The props are accessible as keys of the `this.props` object.

The main difference is the class components can have inner state and life cycle control methods, whereas the functional components are stateless, just rendering stuff. They can be compared to a dotT.js partial.